pkg/kubectl/cmd/create.go

```diff
@@ -172,3 +181,54 @@ func createAndRefresh(info *resource.Info) error {
 	info.Refresh(obj, true)
 	return nil
 }
+
+// nameFromCommandArgs is a utility function for commands that assume the first argument is a resource name
+func nameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) == 0 {
+		return "", cmdutil.UsageError(cmd, "NAME is required")
+	}
+	return args[0], nil
+}
+
+// createSubcommandInternal is intended to be used by subcommands that generator client side objects
+func createSubcommandInternal(f *cmdutil.Factory, cmd *cobra.Command, args []string, out io.Writer, structuredGenerator kubectl.StructuredGenerator) error {
```

Comment:
```
func RunCreate(f *cmdutil.Factory, 
cmd *cobra.Command, 
out io.Writer, 
options *CreateOptions) error {
```

```
func createSubcommandInternal(
f *cmdutil.Factory,
cmd *cobra.Command, 
name string, 
out io.Writer, 
structuredGenerator kubectl.StructuredGenerator) error {
```

I struggle to see how `name` is the tipping point here.  This has the same decoupling as the rest of create flows.